tests/core/test_recovery.py

```diff
@@ -95,6 +97,38 @@ async def test_recovery_ok(tmp_path, user_fs_factory, running_backend, bob):
         assert not um.speculative
 
 
+@pytest.mark.trio
+async def test_recovery_with_wrong_passphrase(
+    tmp_path: Path, user_fs_factory, running_backend, bob: LocalDevice
+):
+    # 1) Create recovery device and export it as file
+
+    recovery_device = await generate_recovery_device(bob)
+
+    assert recovery_device.organization_addr == bob.organization_addr
+    assert recovery_device.device_id != bob.device_id
+    assert recovery_device.device_label != bob.device_label
+    assert recovery_device.human_handle == bob.human_handle
+    assert recovery_device.signing_key != bob.signing_key
+    assert recovery_device.private_key == bob.private_key
+    assert recovery_device.profile == bob.profile
+    assert recovery_device.user_manifest_id == bob.user_manifest_id
+    assert recovery_device.user_manifest_key == bob.user_manifest_key
+    assert recovery_device.local_symkey != bob.local_symkey
+
+    file_name = get_recovery_device_file_name(recovery_device)
+
+    file_path = tmp_path / file_name
+    passphrase = await save_recovery_device(file_path, recovery_device, force=False)
+
+    # 2) Try to load the recovery device file with the wrong passphrase.
+    wrong_passphrase = "-".join([chunk[::-1] for chunk in passphrase.split("-")])
+    assert wrong_passphrase != passphrase
+
+    with pytest.raises(LocalDeviceError):
```

Comment:
`LocalDeviceError` is too broad, it should be a `LocalDeviceCryptoError` specifically.

Reply:
Would be done in another PR see #4048